test/test_prims.py

```diff
@@ -752,6 +759,29 @@ def fn1(x):
             )
             self.assertFalse(includes_aten_to_copy)
 
+    @ops([op for op in op_db if op.name in {"permute"}], allowed_dtypes=(torch.float,))
+    def test_decomposition_method_vararg(self, device, dtype, op):
+        # some ops have vararg variants for the methods. this tests it.
+        # we don't have tests for varargs in OpInfo, so we need to
+        # improvise this a bit.
+        # The rule for general functions (the special cases being e.g. tensor
+        # creation functions taking shapes) is that things can be vararg
+        # if the method has only one argument of sequence type.
+        # we might need to adjust things for the factory functions or
+        # have them do their own test
+        from torch.fx.experimental.proxy_tensor import make_fx
+        from torch._prims.context import TorchRefsMode
+
+        sample_inputs = op.sample_inputs(device, dtype, requires_grad=False)
+        for sample_input in sample_inputs:
+            a = sample_input.input
+            if not sample_input.args[-1]:  # empty tuple cannot be varargs
+                continue
+
+            def f(a, b, c):
+                return getattr(a, op.name)(*b, *c)
```

Comment:
Maybe this could also assert that the results are the same between the op and the vararg variant?

Reply:
Instead of trying to acquire the method from the input, maybe this should just call the method variant of the operation directly? It can be acquired from the OpInfo